client-side/src/components/settings/Preferences.jsx

```diff
@@ -0,0 +1,118 @@
+import React, { useState, useEffect } from 'react';
+import PropTypes from 'prop-types';
+import { useTranslation } from 'react-i18next';
+import moment from 'moment-timezone';
+import { useSelector } from 'react-redux';
+
+import Select from '../../stories/Select/Select.jsx';
+import { selectAuth } from '../../redux/auth/auth.selector.js';
+
+import CONSTANTS from './constantSetting.js';
+import './Preferences.scss';
+
+const createTimeZones = () => {
+  return moment.tz.names().map(timezone => ({
+    value: timezone,
+    text: timezone,
+  }));
+};
+
+const Preferences = ({ onUpdate }) => {
+  const { user } = useSelector(selectAuth);
+  const { t: translate, i18n: localization } = useTranslation();
+  const { LABELS, LANGUAGE, DATE_FORMATS } = CONSTANTS;
+  const { timeZone: initialTimeZone = "UTC", dateFormat: initialDateFormat = DATE_FORMATS[0].value } = user.preference;
```

Comment:
i cant get the UTC from constant because it is one of the options of the timeZone function
Do i need to do :

1. const getDefaultTimeZone = () => {

  const timeZoneOptions = createTimeZones();
  // Here, we simply choose the first time zone as the default.
  // You can apply any other logic to choose a different default.
  return timeZoneOptions.length > 0 ? timeZoneOptions[0].value : 'UTC';
};
.......
const { timeZone: initialTimeZone = getDefaultTimeZone(), dateFormat: initialDateFormat = DATE_FORMATS[0].value } = user.preference;

2. or

// constantSetting.js
const CONSTANTS = {
  DEFAULT_TIME_ZONE: 'UTC' 
};
export default CONSTANTS;
........ timeZone: initialTimeZone = DEFAULT_TIME_ZONE

3. or leave the code as it is?